src/pymovements/gaze/gaze_dataframe.py

```diff
@@ -296,15 +296,22 @@ def split(self, by: Sequence[str]) -> list[GazeDataFrame]:
             A list of new GazeDataFrame instances, each containing a partition of the
             original data with all metadata and configurations preserved.
         """
+        frames = self.frame.partition_by(by=by)
+        # we cannot directly zip because we need to check if by is in events columns
+        events_list = self.events.split(by) if by in self.events.columns else [
+            pm.EventDataFrame(),
```

Comment:
Are you sure this is correct? If `by` is not in the events columns just empty events are created?

Reply:
what else would you propose?

Reply:
if `by` is not included in the events, then log a warning and clone the events to each split.

Reply:
we could even think about restricting it to `trial_columns`, which are used for event extraction?